test/modules/TestResultCoordinator/Reports/EdgeHubRestartTest/EdgeHubRestartMessageReportGenerator.cs

```diff
@@ -0,0 +1,319 @@
+                maxPeriod,
+                medianPeriod,
+                meanPeriod,
+                variancePeriodInMilisec);
+        }
+
+        async Task<(ulong resultCount, bool isNotEmpty, long sequenceNum)> IncrementSequenceNumberAsync(
+            bool isNotEmpty,
+            MoveNextResultAsync MoveNextResultAsync,
+            long targetSequenceNumber,
+            ulong resultCount)
+        {
+            long seqNum = 0;
+
+            while (isNotEmpty && (seqNum < targetSequenceNumber))
+            {
+                (resultCount, isNotEmpty, seqNum) = await MoveNextResultAsync(resultCount);
+            }
+
+            return (resultCount: resultCount,
+                isNotEmpty: isNotEmpty,
+                sequenceNum: seqNum);
+        }
+
+        private async Task<(ulong resultCount, bool hasValue, long sequenceNumber)> MoveNextSenderResultAsync(ulong senderResultCount)
+        {
+            bool hasValue = await this.SenderTestResults.MoveNextAsync();
+            long seqNum = 0;
+            if (hasValue)
+            {
+                senderResultCount++;
+
+                EdgeHubRestartMessageResult senderResult = JsonConvert.DeserializeObject<EdgeHubRestartMessageResult>(this.SenderTestResults.Current.Result);
+                seqNum = this.ParseSenderSequenceNumber(senderResult.SequenceNumber);
+
+                this.AddEntryToCompletedStatusHistogram(senderResult);
+            }
+
+            return (resultCount: senderResultCount,
+                hasValue: hasValue,
+                sequenceNumber: seqNum);
+        }
+
+        private async Task<(ulong resultCount, bool hasValue, long sequenceNumber)> MoveNextReceiverResultAsync(ulong receiverResultCount)
+        {
+            bool hasValue = await this.ReceiverTestResults.MoveNextAsync();
+            long seqNum = 0;
+            if (hasValue)
+            {
+                receiverResultCount++;
+                seqNum = this.ParseReceiverSequenceNumber(this.ReceiverTestResults.Current.Result);
+            }
+
+            return (resultCount: receiverResultCount,
+                hasValue: hasValue,
+                sequenceNumber: seqNum);
+        }
+
+        void ValidateResult(
+            TestOperationResult result,
+            string expectedSource,
+            string testOperationResultType)
+        {
+            if (!result.Source.Equals(expectedSource, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new InvalidDataException($"Result source is '{result.Source}' but expected should be '{expectedSource}'.");
+            }
+
+            if (!result.Type.Equals(testOperationResultType, StringComparison.Ordinal))
+            {
+                throw new InvalidDataException($"Result type is '{result.Type}' but expected should be '{testOperationResultType}'.");
+            }
+        }
+
+        //////////////////////////////////////////////////////////////// HELPER LAND
```

Comment:
Avoid comments like this, pls. The method name should indicate the helper nature of it. Maybe you can make it static.